extensions/amp-story/0.1/bookend/components/article.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AbstractBookendComponent} from './abstract';
+import {dev, user} from '../../../../../src/log';
+import {dict} from '../../../../../src/utils/object';
+import {isProtocolValid, parseUrl} from '../../../../../src/url';
+
+const TAG = 'amp-story';
+
+/**
+ * @typedef {{
+ *   type: string,
+ *   title: string,
+ *   url: string,
+ *   image: (string|undefined)
+ * }}
+ */
+export let BookendArticleComponentDef;
+
+/**
+ * Builder class for the small article component.
+ */
+export class ArticleComponent extends AbstractBookendComponent {
+
+  /** @override */
+  static build(articleJson) {
+    if (!articleJson['title'] || !articleJson['url']) {
+      user().error(TAG,
+          'Articles must contain `title` and `url` fields, skipping invalid.');
+      return null;
+    }
+
+    user().assert(isProtocolValid(articleJson['url']),
+        `Unsupported protocol for article URL ${articleJson['url']}`);
+
+    const article = {
+      type: 'small',
+      title: dev().assert(articleJson['title']),
+      url: dev().assert(articleJson['url']),
+      domainName: parseUrl(dev().assert(articleJson['url'])).hostname,
+    };
+
+    if (articleJson['image']) {
+      user().assert(isProtocolValid(articleJson['image']),
+          `Unsupported protocol for article image URL ${articleJson['image']}`);
+      article.image = dev().assert(articleJson['image']);
+    }
+
+    return /** @type {!RelatedArticleDef} */ (article);
+  }
+
+  /** @override */
+  static buildTemplate(articleData) {
```

Comment:
We want to migrate away from using simple templates and instead use the static HTML template helper where possible: #14659